osquery/main/tests.cpp

```diff
@@ -1,20 +1,124 @@
 /*
- *  Copyright (c) 2014-present, Facebook, Inc.
- *  All rights reserved.
- *
- *  This source code is licensed under the BSD-style license found in the
- *  LICENSE file in the root directory of this source tree. An additional grant
- *  of patent rights can be found in the PATENTS file in the same directory.
- *
- */
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#ifdef WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Shlwapi.h>
+#endif
+
+#include <memory>
 
 #include <gtest/gtest.h>
 
 #include <osquery/logger.h>
 
+#include "osquery/core/process.h"
 #include "osquery/core/test_util.h"
 
+#define OSQUERY_TESTS_MODULENAME  "osquery_tests.exe"
+
+char *self_exec_path = nullptr;
```

Comment:
Got any prefs on where to place the header file? I'm thinking `osquery/core/testing.h`?